internal/venice-client-common/src/main/java/com/linkedin/venice/stats/metrics/MetricEntityStateOneEnum.java

```diff
@@ -0,0 +1,143 @@
+  private MetricEntityStateOneEnum(
+      MetricEntity metricEntity,
+      VeniceOpenTelemetryMetricsRepository otelRepository,
+      Map<VeniceMetricsDimensions, String> baseDimensionsMap,
+      Class<E> enumTypeClass) {
+    this(metricEntity, otelRepository, null, null, Collections.EMPTY_LIST, baseDimensionsMap, enumTypeClass);
+  }
+
+  /** should not be called directly, call {@link #create} instead */
+  private MetricEntityStateOneEnum(
+      MetricEntity metricEntity,
+      VeniceOpenTelemetryMetricsRepository otelRepository,
+      TehutiSensorRegistrationFunction registerTehutiSensorFn,
+      TehutiMetricNameEnum tehutiMetricNameEnum,
+      List<MeasurableStat> tehutiMetricStats,
+      Map<VeniceMetricsDimensions, String> baseDimensionsMap,
+      Class<E> enumTypeClass) {
+    super(metricEntity, otelRepository, registerTehutiSensorFn, tehutiMetricNameEnum, tehutiMetricStats);
+    validateRequiredDimensions(metricEntity, baseDimensionsMap, enumTypeClass);
+    this.enumTypeClass = enumTypeClass;
+    this.attributesEnumMap = new EnumMap<>(enumTypeClass);
+    if (emitOpenTelemetryMetrics()) {
+      createAttributesEnumMap(metricEntity, otelRepository, baseDimensionsMap);
+    }
+  }
+
+  /** Factory method with named parameters to ensure the passed in enumTypeClass are in the same order as E */
+  public static <E extends Enum<E> & VeniceDimensionInterface> MetricEntityStateOneEnum<E> create(
+      MetricEntity metricEntity,
+      VeniceOpenTelemetryMetricsRepository otelRepository,
+      Map<VeniceMetricsDimensions, String> baseDimensionsMap,
+      Class<E> enumTypeClass) {
+    return new MetricEntityStateOneEnum<>(metricEntity, otelRepository, baseDimensionsMap, enumTypeClass);
+  }
+
+  /** Overloaded Factory method for constructor with Tehuti parameters */
+  public static <E extends Enum<E> & VeniceDimensionInterface> MetricEntityStateOneEnum<E> create(
+      MetricEntity metricEntity,
+      VeniceOpenTelemetryMetricsRepository otelRepository,
+      TehutiSensorRegistrationFunction registerTehutiSensorFn,
+      TehutiMetricNameEnum tehutiMetricNameEnum,
+      List<MeasurableStat> tehutiMetricStats,
+      Map<VeniceMetricsDimensions, String> baseDimensionsMap,
+      Class<E> enumTypeClass) {
+    return new MetricEntityStateOneEnum<>(
+        metricEntity,
+        otelRepository,
+        registerTehutiSensorFn,
+        tehutiMetricNameEnum,
+        tehutiMetricStats,
+        baseDimensionsMap,
+        enumTypeClass);
+  }
+
+  private void createAttributesEnumMap(
+      MetricEntity metricEntity,
+      VeniceOpenTelemetryMetricsRepository otelRepository,
+      Map<VeniceMetricsDimensions, String> baseDimensionsMap) {
+    Map<VeniceMetricsDimensions, String> additionalDimensionsMap = new HashMap<>();
+    for (E enumValue: enumTypeClass.getEnumConstants()) {
+      additionalDimensionsMap.put(enumValue.getDimensionName(), enumValue.getDimensionValue());
```

Comment:
Is it the case that all `enumValue` share the same `getDimensionName()` and because of that, in fact, later `enumValue.getDimensionName()` in iteration always overwrites the earlier ones?